v2/apiserver/internal/system/rest/healthcheck_endpoints.go

```diff
@@ -0,0 +1,95 @@
+package rest
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/queue"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/restmachinery"
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// HealthcheckEndpoints implements restmachinery.Endpoints to provide
+// Healthcheck-related URL --> action mappings to a restmachinery.Server.
+type HealthcheckEndpoints struct {
+	APIServerVersion string
+	DatabaseClient   *mongo.Client
+	WriterFactory    queue.WriterFactory
+}
+
+// Register is invoked by restmachinery.Server to register Healthcheck-related
+// URL --> action mappings to a restmachinery.Server.
+func (h *HealthcheckEndpoints) Register(router *mux.Router) {
+	// Get /healthz
+	router.HandleFunc(
+		"/healthz",
+		h.healthz,
+	).Methods(http.MethodGet)
+
+	// Get /ping
+	router.HandleFunc(
+		"/ping",
+		h.ping,
+	).Methods(http.MethodGet)
+}
+
+// healthz is the main healthcheck endpoint for the api server.  The endpoint
+// logic verifies connectivity to crucial dependencies, e.g. the database
+// and messaging queues.
+func (h *HealthcheckEndpoints) healthz(w http.ResponseWriter, r *http.Request) {
+	restmachinery.ServeRequest(
+		restmachinery.InboundRequest{
+			W: w,
+			R: r,
+			EndpointLogic: func() (interface{}, error) {
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+				defer cancel()
+
+				// Check Database
+				err := h.DatabaseClient.Ping(ctx, nil)
+				if err != nil {
+					return false, errors.New("error communicating with database")
+				}
+
+				// Check messaging queue
+				writer, err := h.WriterFactory.NewWriter("healthz")
+				if err != nil {
+					return false, errors.New("error instantiating messaging queue writer")
+				}
+				defer writer.Close(ctx)
+				if err := writer.Write(
+					ctx,
+					"ping",
+					&queue.MessageOptions{},
+				); err != nil {
+					return false, errors.New("error writing to messaging queue")
+				}
+
+				return []byte("ok"), nil
```

Comment:
This may require some more research. I have a vague recollection that k8s prefers heath checks to return an empty JSON object. A quick search isn't uncovering anything that backs that up, but I swear I've read that _somewhere_. I'll keep looking.

Reply:
Interestingly, performing a GET on the `/healthz` endpoint of the K8s api server's REST client  - at least via `DoRaw()` - returns `[]byte("ok")` if all is well (Discovered this via the recent addition of adding such a check to the observer)